provider/core/src/resource.rs

```diff
@@ -14,17 +14,6 @@ use core::fmt::Write;
 use icu_locid::LanguageIdentifier;
 use writeable::{LengthHint, Writeable};
 
-/// A compact hash of a [`ResourceKey`]. Useful for keys in maps.
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Hash)]
-#[repr(transparent)]
-pub struct ResourceKeyHash([u8; 4]);
-
-impl ResourceKeyHash {
-    const fn compute_from_str(path: &str) -> Self {
-        Self(helpers::fxhash_32(path.as_bytes()).to_le_bytes())
-    }
-}
```

Comment:
I got rid of this because to use it as a `ZeroMap2d` key it needs to be `AsULE`, `ZeroMapKV`, `Serialize` and `Deserialize`. Implementing those here would introduce a fairly strong tie to `serde` and `zerovec`, and also a lot of boilerplate. `u32` on the other hand implements all those traits already, and we don't need to enforce the little-endian encoding here, as that's handled by `u32`'s `ULE` implementation.

Reply:
There are two topics here: unwrapping the primitive, and switching fro `[u8; 4]` to `u32`.

On the first: I'm torn.  In general I tend to prefer to use Rust's newtype idiom to wrap all primitives, simply because it gives us better type safety with a zero-cost abstraction.  All of the impls you mentioned are just a few lines of code.

On the second: I generally would prefer moving forward to use fully ULE types where possible (those with `AsULE::ULE = Self`).  The nice thing about using fully ULE types everywhere is that you can easily combine and nest things into other fully ULE types.

Both of these positions are debatable.

Reply:
How about if we do the following:

1. Restore `ResourceKeyHash` to what it was before
2. Add a getter to `ResourceKeyHash` that returns the inner type (`[u8; 4]`)
3. Keep `ZeroMap2d` as you have it specified in this PR
4. In the `get_file` functions, convert the `[u8; 4]` to `u32` and use it

This makes the specific decision here an implementation detail that we can work out in the future if needed.

Reply:
If we're using `[u8; 4]` already we should make that the key type (instead of converting to `u32` whose `ULE` converts it back). 
However arrays don't currently implement `ULE`, and my attempt seems to invalidate some invariant.